doc/arkode/guide/source/Usage/SPRKStep_c_interface/User_callable.rst

```diff
@@ -0,0 +1,1229 @@
+
+   :param arkode_mem: pointer to the SPRKStep memory block.
+   :param itype: requested interpolant type (``ARK_INTERP_HERMITE`` or ``ARK_INTERP_LAGRANGE``)
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the SPRKStep memory is ``NULL``
+      * *ARK_MEM_FAIL* if the interpolation module cannot be allocated
+      * *ARK_ILL_INPUT* if the *itype* argument is not recognized or the
+        interpolation module has already been initialized
+
+   **Notes:**
+      The Hermite interpolation module is described in
+      :numref:`ARKODE.Mathematics.Interpolation.Hermite`, and the Lagrange interpolation module
+      is described in :numref:`ARKODE.Mathematics.Interpolation.Lagrange`.
+
+      This routine frees any previously-allocated interpolation module, and re-creates
+      one according to the specified argument.  Thus any previous calls to
+      :c:func:`SPRKStepSetInterpolantDegree()` will be nullified.
+
+      This routine must be called *after* the call to :c:func:`SPRKStepCreate`.
+      After the first call to :c:func:`SPRKStepEvolve()` the interpolation type may
+      not be changed without first calling :c:func:`SPRKStepReInit()`.
+
+      If this routine is not called, the Lagrange interpolation module will be used.
+      Our testing indicates that Lagrange interpolation does a better job of conserving
+      quantites than Hermite interpolation.
+
+
+
+.. c:function:: int SPRKStepSetInterpolantDegree(void* arkode_mem, int degree)
+
+   Specifies the degree of the polynomial interpolant
+   used for dense output (i.e. interpolation of solution output values
+   and implicit method predictors).
+
+   :param arkode_mem: pointer to the SPRKStep memory block.
+   :param degree: requested polynomial degree.
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the SPRKStep memory or interpolation module are ``NULL``
+      * *ARK_INTERP_FAIL* if this is called after :c:func:`SPRKStepEvolve()`
+      * *ARK_ILL_INPUT* if an argument has an illegal value or the
+        interpolation module has already been initialized
+
+   **Notes:**
+      Allowed values are between 0 and 5.
+
+      This routine should be called *after* :c:func:`SPRKStepCreate` and *before*
+      :c:func:`SPRKStepEvolve()`. After the first call to :c:func:`SPRKStepEvolve()`
+      the interpolation degree may not be changed without first calling
+      :c:func:`SPRKStepReInit()`.
+
+      If a user calls both this routine and :c:func:`SPRKStepSetInterpolantType()`, then
+      :c:func:`SPRKStepSetInterpolantType()` must be called first.
+
+      Since the accuracy of any polynomial interpolant is limited by the
+      accuracy of the time-step solutions on which it is based, the *actual*
+      polynomial degree that is used by SPRKStep will be the minimum of
+      :math:`q-1` and the input *degree*, for :math:`q > 1` where :math:`q` is
+      the order of accuracy for the time integration method.
```

Comment:
Is this addition from a recent change?

Reply:
Yes, this was something I added recently in ARKStep and ERKStep where we were getting unexpected results comparing to a native IMEX Euler implementation because of interpolation.